### ISSUE TITLE
How to customize different fine-tuning task like protein-protein interaction (PPI) ?

### ISSUE THREAD

**Issue by zrhsu0911** [nadavbra/protein_bert#34]
Hello:
As mentioned in the title, I wonder how to customize fine-tuning task like protein-protein interaction? I am not sure which part of finetuning.py (or perhaps FinetuningModelGenerator in model_generation.py) should I change; I checked all fine-tuning task in this repo, always one protein seq in one single sample.
The fellowing CSV format is the dataset I want to use in my task. Many thanks.

|label|seq_A                                                                 |seq_B                                                                 |
|-----|----------------------------------------------------------------------|----------------------------------------------------------------------|
|0    |MAVSVTPIRDTKWLTLEVCREFQRGTCSRPDTECKFAHPSKSCQVENGRVIACFDSLKGRCSRENCKYLH|MATTNSFTILIFMILATTSSTFATLGEMVTVLSIDGGGIKGIIPATILEFLEGQLQEVDNNTDARLADYF|
|1    |MCCEKWSRVAEMFLFIEEREDCKILCLCSRAFVEDRKLYNLGLKGYYIRDSGNNSGDQATEEEEGGYSCG|MSASSRFIPEHRRQNYKGKGTFQADELRRRRETQQIEIRKQKREENLNKRRNLVDVQEPAEETIPLEQDK|

**Comment by nadavbra** [comment]
@zrhsu0911 If I understand correctly the task you are interested in (taking two protein sequences as input and predicting whether or not they should bind by outputing a single probability) - it's not supported out of the box by the existing ProteinBERT package. You will have to use ProteinBERT to create your own architecture with keras. For example, you could make a model comprised of two instances of ProteinBERT, each taking a separate sequence as input, and then using the final layer's global embeddings for these two sequences and add a final dense layer that decides whether or not they bind based on these representations (and fine-tune ProteinBERT's pretrained weights on your data).

**Comment by zrhsu0911** [comment]
> @zrhsu0911 If I understand correctly the task you are interested in (taking two protein sequences as input and predicting whether or not they should bind by outputing a single probability) - it's not supported out of the box by the existing ProteinBERT package. You will have to use ProteinBERT to create your own architecture with keras. For example, you could make a model comprised of two instances of ProteinBERT, each taking a separate sequence as input, and then using the final layer's global embeddings for these two sequences and add a final dense layer that decides whether or not they bind based on these representations (and fine-tune ProteinBERT's pretrained weights on your data).

**Comment by zrhsu0911** [comment]
@nadavbra
Thank you for your swift response to my question, maybe I can just use this only for embeddings.

**Comment by nadavbra** [comment]
Yes, you can definitely just use ProteinBERT for extracting embeddings and then use a simple ML algorithm. That would be the simplest approach.

**Comment by yelou2022** [comment]
@nadavbra Hi, 
I have a small question to ask. In the context of the downstream protein-protein interaction (PPI) task, which benchmark dataset should be chosen for fine-tuning? After browsing through the benchmark datasets provided, I found that only ProFET_NP_SP and signalp_binary have label data in a format consistent with the PPI task (label: 0 or 1). If we were to perform fine-tuning, which dataset should we choose? Or, in your and your team's opinion, would it be better to use the fine-tuned model or directly use the pre-trained model provided by you?

best regard!

**Comment by nadavbra** [comment]
We don't have a benchmark for PPI in ProteinBERT (we never tested that in our paper) - that's a general bioinformatics question that is outside my area of expertise.
I'm not sure I understand the question about pre-trained vs. fine-tuned model. Fine-tuned for what? We only provide the pre-trained model which I expect would be useful for this task after fine-tuning.

**Comment by yelou2022** [comment]
Sorry, I may not have explained it clearly. My intention is to use the ProteinBERT model to extract protein features and improve the accuracy of protein-protein interaction (PPI) predictions. However, I am unsure whether it would be better to directly use the pre-trained model provided by your team or fine-tune ProteinBERT before utilizing it. If I want to fine-tune ProteinBERT, but your team does not provide a benchmark dataset for PPI fine-tuning, it hinders the progress of my next steps. Do you have any suggestions regarding this issue?

Wish you a pleasant day!
